### PR TITLE
Member Submission of Attendance

### LINKED ISSUE #107: Member Submitted Attendance for Meetings and Seminars
Members should be able to report attendance for a meeting for approval by an Eboard member. 

From a UI standpoint we could just prepend unapproved submissions to the top of the history page and give options for Eboard members to approve or deny. As far as schema goes, we just need to add a boolean column to the CommitteeMeeting table to represent each meeting's status.

(We also need to check for only approved meetings when calculating number of CMs attended.)

### PR DESCRIPTION
**Pull request by mbillow**
Fixes #107 

Allows members to submit attendance for Committee Meetings and Seminars. This attendance is then added to a queue for Eboard Approval.

screen shot 2017-01-22 at 16 49 20

### CODE REVIEW

File: conditional/blueprints/attendance.py [hunk 1]
```diff
@@ -182,9 +172,8 @@ def submit_committee_attendance():
 
     user_name = request.headers.get('x-webauth-user')
     account = ldap_get_member(user_name)
-    if not ldap_is_eboard(account):
-        return "must be eboard", 403
-
+    approved = ldap_is_eboard(account)
+    print(str(approved) + "IS AN EBAORD MEMBER")
```
Comment (liam-middlebrook) [inline conditional/blueprints/attendance.py]: EVAULATIONS

File: conditional/blueprints/attendance.py [hunk 2]
```diff
@@ -383,7 +371,7 @@ def alter_house_excuse(uid, hid):
 def attendance_history():
 
 
-    def get_meeting_attendees(meeting_id):
+    def get_comm_attendees(meeting_id):
```
Comment (liam-middlebrook) [inline conditional/blueprints/attendance.py]: we should really change this to use the term directorship eventually

File: conditional/blueprints/attendance.py [hunk 3]
```diff
@@ -395,6 +383,18 @@ def get_meeting_attendees(meeting_id):
                              FreshmanAccount.id == freshman).first().name)
         return attendees
 
+    def get_sem_attendees(meeting_id):
```
Comment (liam-middlebrook) [inline conditional/blueprints/attendance.py]: I really don't think we need to abbreviate function names

File: conditional/blueprints/attendance.py [hunk 4]
```diff
@@ -459,12 +476,87 @@ def alter_committee_attendance(cid):
     db.session.commit()
     return jsonify({"success": True}), 200
 
-@attendance_bp.route('/attendance/cm/<cid>', methods=['GET', 'DELETE'])
+
+@attendance_bp.route('/attendance/alter/ts/<cid>', methods=['POST'])
```
Comment (liam-middlebrook) [inline conditional/blueprints/attendance.py]: cid only made sense when this was referring to committee meetings

File: conditional/blueprints/dashboard.py [hunk 5]
```diff
@@ -58,7 +58,8 @@ def display_dashboard():
     c_meetings = [m.meeting_id for m in
                   MemberCommitteeAttendance.query.filter(
                       MemberCommitteeAttendance.uid == member.uid
-                  )]
+                  ) if CommitteeMeeting.query.filter(
```
Comment (liam-middlebrook) [inline conditional/blueprints/dashboard.py]: we have a bunch of oddities for SQL queries, but I think we could do a fancy statement with a single query.

Comment (mbillow) [inline conditional/blueprints/dashboard.py]: I am not sure the sacrifice in readability is worth the four or five milliseconds you would save on load time. I looked back at the query I wrote for the housing queue and had to turn my monitor upside just trying to remember why I did what I did.

File: conditional/templates/attendance_history.html [hunk 6]
```diff
@@ -4,6 +4,60 @@
 {% endblock %}
 {% block body %}
 <div class="container main">
+    {% if pending_cm|length != 0 and current_page == 1 %}
+    <h3 class="page-title">Pending Submittions</h3>
```
Comment (csssuf) [inline conditional/templates/attendance_history.html]: Submissions.

File: conditional/templates/attendance_history.html [hunk 7]
```diff
@@ -81,4 +135,29 @@ <h4 class="modal-title" id="editMeetingTitle">Edit Meeting</h4>
...
     </div>
 </div>
+
+<div class="modal fade" id="reviewMeeting" tabindex="-1">
+    <div class="vertical-alignment-helper">
+        <div class="modal-dialog vertical-align-center">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
+                    <h4 class="modal-title" id="editMeetingTitle">Review Submittion</h4>
```
Comment (csssuf) [inline conditional/templates/attendance_history.html]: Submission.

**Comment by mbillow** [comment]
@csssuf I code good... but spell good is hard.

**Comment by mbillow** [comment]
Has anyone had a chance to test this?

**Pull request merged**